tests/Model/Changelog/ChangelogItemTest.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace SilverStripe\Cow\Tests\Model\Changelog;
+
+use Gitonomy\Git\Commit;
+use PHPUnit_Framework_TestCase;
+use SilverStripe\Cow\Model\Changelog\ChangelogItem;
+use SilverStripe\Cow\Model\Changelog\ChangelogLibrary;
+
+class ChangelogItemTest extends PHPUnit_Framework_TestCase
+{
+    /**
+     * @var ChangelogLibrary
+     */
+    protected $library;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->library = $this->getMockBuilder(ChangelogLibrary::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+    }
+
+    /**
+     * @param string $message
+     * @param string $expected
+     * @dataProvider typeProvider
+     */
+    public function testGetType($message, $expected)
+    {
+        $commit = $this->createMock(Commit::class);
+        $commit->expects($this->atLeastOnce())->method('getSubjectMessage')->willReturn($message);
+
+        $item = new ChangelogItem($this->library, $commit);
+        $this->assertSame($expected, $item->getType());
+    }
+
+    /**
+     * @return array[]
+     */
+    public function typeProvider()
+    {
+        return [
+            ['BUG Fixed some regex rules', 'Bugfixes'],
+            ['FIX Fixed some regex rules', 'Bugfixes'],
+            ['Fixed some regex rules', 'Bugfixes'],
+            ['Fixing some regex rules', 'Bugfixes'],
+            ['API Remove DataObject', 'API Changes'],
+            ['API CHANGE Remove DataObject', 'API Changes'],
+            ['NEW Allow Python transpilation', 'Features and Enhancements'],
+            ['ENH Support goland PHP extension for mad caching', 'Features and Enhancements'],
+            ['[SS-2047-123] Prevent possible doubt with cow test coverage', 'Security'],
+            ['[ss-2047-123] Prevent possible doubt with cow test coverage', 'Security'],
+            ['Default fallback doesn\'t categorise commit', null],
+        ];
+    }
+
+    /**
+     * @param string $message
+     * @param string $expected
+     * @dataProvider shortMessageProvider
+     */
+    public function testGetShortMessage($message, $expected)
+    {
+        $commit = $this->createMock(Commit::class);
+        $commit->expects($this->once())->method('getSubjectMessage')->willReturn($message);
+
+        $item = new ChangelogItem($this->library, $commit);
+        $this->assertSame($expected, $item->getShortMessage());
+    }
+
+    /**
+     * @return array[]
+     */
+    public function shortMessageProvider()
+    {
+        return [
+            ['BUG Entwine shouldn\'t use jQuery', 'Entwine shouldn\'t use jQuery'],
```

Comment:
This generator uses different messages to the other which means I can't be sure they act as expected across both tests